tests/checks/statistical/test_categorical_statistical.py

```diff
@@ -0,0 +1,31 @@
+import unittest
+
+import pandas as pd
+
+from shift_detector.Detector import Detector
+from shift_detector.checks.statistical_checks.CategoricalStatisticalCheck import chi2_test, CategoricalStatisticalCheck
+
+
+class TestCategoricalStatisticalCheck(unittest.TestCase):
+
+    def test_chi2_test_result(self):
+        part1 = pd.Series((['severe reaction'] * 29) + (['no severe reaction'] * 4757), name='thigh')
+        part2 = pd.Series((['severe reaction'] * 75) + (['no severe reaction'] * 8839), name='arm')
+        p = chi2_test(part1, part2)
+        self.assertAlmostEqual(0.17471089, p)
+
+    def test_not_significant(self):
+        df1 = pd.DataFrame.from_dict({'vaccination_reaction': (['severe reaction'] * 29) + (['no severe reaction'] * 4757)})
+        df2 = pd.DataFrame.from_dict({'vaccination_reaction': (['severe reaction'] * 75) + (['no severe reaction'] * 8839)})
+        detector = Detector(df1=df1, df2=df2)
```

Comment:
You should create a Store and call the Check directly instead of the Detector in order to not be depended of the detector. In addition, there should be one test, that conformance with the detector.